data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/ConsumerVerticleFactoryImpl.java

```diff
@@ -171,7 +153,9 @@ public AbstractVerticle get(final DataPlaneContract.Resource resource, final Dat
           new AttributesFilter(egress.getFilter().getAttributesMap()) :
           Filter.noop();
 
-        final var responseHandler = getNoopResponseHandlerOrDefault(egress, () -> getKafkaResponseHandler(vertx, producerConfigs, resource));
+        final var responseHandler = getResponseHandler(egress,
+          () -> getResponseToKafkaTopicHandler(vertx, producerConfigs, resource),
+          () -> getResponseToHttpEndpointHandler(vertx, egress.getReplyUrl()));
```

Comment:
Can't we do this?

```suggestion
          () -> new ResponseToHttpEndpointHandler(createConsumerRecordSender(vertx, egress.getReplyUrl(), egressConfig));
```

Reply:
Then we can remove the `getResponseToHttpEndpointHandler` method or wrap the suggested logic in that method.

Reply:
Yes, thanks for catching that up. I forgot about my TODOs I added there to do something similar